tests/test_project.py

```diff
@@ -177,6 +177,29 @@ def test_generate_no_handlers(project):
     mock_plugin.generate.assert_called_once_with(project)
 
 
+def test_generate_no_type_defined(project, tmp_path_factory):
```

Comment:
this looks like the exact same test as below... can we use `@parametrized`?
http://doc.pytest.org/en/latest/example/parametrize.html

Reply:
Sure